app/src/main/java/org/sagebionetworks/research/mpower/tracking/TrackingMenuFragment.java

```diff
@@ -145,16 +138,14 @@ public void onCreate(@Nullable Bundle savedInstanceState) {
         AndroidSupportInjection.inject(this);
         showTracking = getResources().getBoolean(R.bool.show_data_tracking);
         trackingMenuViewModel =
-                ViewModelProviders.of(this).get(TrackingMenuFragmentViewModel.class);
-        if (showTracking) {
-            trackingMenuViewModel.setSelectedId(R.id.tracking_tab);
-        } else {
+                ViewModelProviders.of(getActivity()).get(TrackingMenuFragmentViewModel.class);
```

Comment:
It's also important to attach the view models to the activity here instead of the fragment, as nested child fragments are appropriately destroyed more frequently than your average fragment.